### ISSUE TITLE
Implement TransformStream (Web Stream) for csv-parse

### ISSUE THREAD

**Issue by uasan** [adaltas/node-csv#444]
**Summary**

Hi, our data source is ReadableStream, due to performance we would not like to use the `twoWeb` or `fromWeb` adapters, I see in the parser there is a file
https://github.com/adaltas/node-csv/blob/master/packages/csv-parse/lib/stream.js
but it is not exported and the implementation itself is not complete, there is no error handling and no promise waiting for the chunk parsing to complete.

I can make a PR in which I will add everything necessary for its correct operation to the stream.js file and make an export, would you be against this PR?

**Motivation**

If the data source is a web stream, such as fetch, then a productive pipe interface is needed to parse the CSV.

**Alternative**

I can implement TransformStream in my application if I export from your library the `transform` function from the file https://github.com/adaltas/node-csv/blob/master/packages/csv-parse/lib/api/index.js

**Comment by wdavidw** [comment]
You are right, the Web Stream implementation is an undocumented API. I did it some time ago but in my memory it was functional. Errors are expected to be propagated but maybe you have something in mind that must be handle inside the transformer. I did some performance comparison and it was slower than with the Node.js stream API. I would review PR, just that the test coverage need to be complete.

**Comment by uasan** [comment]
PR: https://github.com/adaltas/node-csv/pull/445

Yes, you are right, parser errors are always sync and web streams will handle them, I made all `transform` and `flush` synchronous, they will not create extra promise objects, I also moved callbacks to closures so as not to create new functions for each chunk.

**Comment by uasan** [comment]
I added errors handle to TransformStream

**Comment by uasan** [comment]
@wdavidw When is the new version planned to be published in NPM?
Thanks.

**Comment by wdavidw** [comment]
I am fighting with the latest lerna version and its `version` command which requires a package lock file which the CI doesn't really appreciate.

**Comment by uasan** [comment]
I read the specification again and did some experiments, for correct error handling it is enough to call the function `controller.error(error)`, throwing an exception can be removed, it is unnecessary.

Waiting for the release of a new version )

**Comment by wdavidw** [comment]
Indeed, the "API Web Stream/stream web TransformStream" test behaves the same.

**Comment by uasan** [comment]
By the way, it will be useful for many if the documentation includes descriptions and a simple example of a web stream, I think many will be interested in such an example:

```JS
import { parse  } from 'csv-parse/stream';

const response = await fetch('file.csv');

for await (const record of response.body.pipeThrough(parse())) {
   console.log(record);
}

```

**Comment by wdavidw** [comment]
It certainly makes sense. I was waiting for more knowledge/confidence in the implementation before pushing docs and sample. Also lack of time..

**Comment by wdavidw** [comment]
Version 5.6.0 of csv-parse is published.

**Comment by uasan** [comment]
Thanks!

**Comment by wdavidw** [comment]
What do you mean, `import { parse } from 'csv-parse/stream';` shall work with Node.js
